src/dex/fluid-dex/fluid-dex.ts

```diff
@@ -198,26 +191,28 @@ export class FluidDex extends SimpleExchange implements IDex<FluidDexData> {
       // Make sure the pool meets the optional limitPools filter.
       if (limitPools && !limitPools.includes(pool.id)) return null;
 
-      const eventPool = this.eventPools[pool.id];
-
-      if (!eventPool) {
-        this.logger.error(`fluid-dex pool ${pool.id}: No EventPool found.`);
+      const liquidityProxyState = this.liquidityProxy.getState(blockNumber);
```

Comment:
Maybe better `getStateOrGenerate()`? Don't really see in what cases it can worse anything (it can overuse RPC calls instead of return empty prices when state is null, but to detect this better adding log that we generating state)